android/app/src/main/java/com/vithop/tictactoe/Status.kt

```diff
@@ -1,25 +1,26 @@
 package com.vithop.tictactoe
 
-sealed class Status {
-  object XTurn : Status()
-  object OTurn : Status()
-  data class XWon(val winPositions: Bitboard) : Status() {
-    override fun toString() = super.toString()
-  }
-
-  data class OWon(val winPositions: Bitboard) : Status() {
-    override fun toString() = super.toString()
-  }
+enum class Status {
+  XTurn,
+  OTurn,
+  XWon,
+  OWon,
+  Draw;
 
-  object Draw : Status()
+  private var winPositions = Bitboard()
 
-  override fun toString() = when (this) {
-    XTurn -> "X turn"
-    OTurn -> "O turn"
-    is XWon -> "X won"
-    is OWon -> "O won"
-    Draw -> "Draw"
+  fun getWinPosition(): Bitboard {
+    isWinState()
+    return this.winPositions
   }
 
   val isGameOver get() = this != XTurn && this != OTurn
-}
\ No newline at end of file
+
+  val isWinState = { check(this == XWon || this == OWon) { "Invalid status: $this" } }
+
+  companion object {
+    val xWon = { winPos: Bitboard -> XWon.apply { this.winPositions = winPos } }
```

Comment:
Maybe function syntax is cleaner?
```suggestion
    fun XWon(winPositions: Bitboard) = XWon.apply{ this.winPositions = winPositions }
```